lib/type/number.ex

```diff
@@ -11,9 +11,118 @@ defmodule Litmus.Type.Number do
   ]
 
   @type t :: %__MODULE__{
-          min: non_neg_integer,
-          max: non_neg_integer,
+          min: number | nil,
+          max: number | nil,
           integer: boolean,
           required: boolean
         }
+
+  alias Litmus.Required
+
+  @spec validate_field(t, binary, map) :: {:ok, map} | {:error, binary}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- convert(type, field, data),
+         {:ok, data} <- min_validate(type, field, data),
+         {:ok, data} <- max_validate(type, field, data),
+         {:ok, data} <- integer_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec string_to_number(binary) :: number | nil
+  defp string_to_number(str) do
+    str = if String.starts_with?(str, "."), do: "0" <> str, else: str
+
+    cond do
+      !Regex.match?(@number_regex, str) -> nil
```

Comment:
I don't think this check is necessary. It should be sufficient to use your `string_to_integer` and `string_to_float` functions, so can we just leave this out?